packages/zen-bull-netting/src/ZenBullNetting.sol

```diff
@@ -862,28 +789,26 @@ contract ZenBullNetting is Ownable, EIP712, FlashSwap {
         isAuctionLive = false;
     }
 
-    /**
-     * @notice uniswap flash swap callback function to handle different types of flashswaps
-     * @dev this function will be called by flashswap callback function uniswapV3SwapCallback()
-     * @param _uniFlashSwapData UniFlashswapCallbackData struct
-     */
-    function _uniFlashSwap(UniFlashswapCallbackData memory _uniFlashSwapData) internal override {
-        if (FLASH_SOURCE(_uniFlashSwapData.callSource) == FLASH_SOURCE.WITHDRAW_AUCTION) {
-            WithdrawAuctionData memory data =
-                abi.decode(_uniFlashSwapData.callData, (WithdrawAuctionData));
-
-            IZenBullStrategy(zenBull).withdraw(data.zenBullAmountToBurn);
-            IWETH(weth).deposit{value: _uniFlashSwapData.amountToPay}();
-            IWETH(weth).transfer(_uniFlashSwapData.pool, _uniFlashSwapData.amountToPay);
-        } else if (FLASH_SOURCE(_uniFlashSwapData.callSource) == FLASH_SOURCE.DEPOSIT_AUCTION) {
-            DepositAuctionData memory data =
-                abi.decode(_uniFlashSwapData.callData, (DepositAuctionData));
+    function _uniFlashSwap(
+        address pool,
+        uint256 amountToPay,
+        bytes memory callData,
+        uint8 callSource
+    ) internal override {
+        if (callSource == 0) {
+            uint256 zenBullAmountToBurn = abi.decode(callData, (uint256));
+
+            IZenBullStrategy(zenBull).withdraw(zenBullAmountToBurn);
+            IWETH(weth).deposit{value: amountToPay}();
+            IWETH(weth).transfer(pool, amountToPay);
+        } else if (callSource == 1) {
+            uint256 wethToLend = abi.decode(callData, (uint256));
 
             IWETH(weth).withdraw(IWETH(weth).balanceOf(address(this)));
-            IZenBullStrategy(zenBull).deposit{value: data.wethToLend}(
+            IZenBullStrategy(zenBull).deposit{value: wethToLend}(
                 IERC20(crab).balanceOf(address(this))
```

Comment:
is it possible you get a state here where we have more crab then expected, which means that we deposit too much crab.  Lets say someone sends 1 wei to the contract and then you try to deposit with 1 wei more than you sent as crab param, that means you won't have enough eth sent I think?

Reply:
I don't think so...